src/components/DiagramContainer.tsx

```diff
@@ -93,28 +64,174 @@ class DiagramContainer extends React.Component<DiagramProps, DiagramState> {
+        }
+      }
+    }
+
+    // node removed
+    if (removedNodeKeys) {
+      for (let removedNodeKey of removedNodeKeys) {
+        if (removedNodeKey === undefined) {
+          continue;
+        }
+
+        fetch(
+          "https://localhost:7009/graph/vertices?" +
+            new URLSearchParams({
+              key: removedNodeKey.toString(),
+            }),
+          { method: "DELETE" }
+        )
+          .then((response) => {
+            if (response.ok) {
+              console.log(
+                `Successfully deleted key ${removedNodeKey} on backend`
+              );
+            } else {
+              throw new Error(
+                JSON.stringify({
+                  status: response.status,
+                  body: response.text(),
+                })
+              );
+            }
+          })
+          .catch((err) => console.log(err));
+      }
+    }
+
+    if (modifiedLinkData) {
+      console.log(modifiedLinkData);
+
+      for (let modifiedLink of modifiedLinkData) {
+        if (
+          insertedLinkKeys &&
+          insertedLinkKeys.includes(modifiedLink["key"])
+        ) {
+          fetch(
+            "https://localhost:7009/graph/edges?" +
+              new URLSearchParams({
+                srcKey: modifiedLink["from"].toString(),
+                dstKey: modifiedLink["to"].toString(),
+              }),
+            { method: "POST" }
+          )
+            .then((response) => {
+              if (response.ok) {
+                console.log(`Successfully added edge on backend`);
+                console.log(modifiedLink);
+              } else {
+                throw new Error(
+                  JSON.stringify({
+                    status: response.status,
+                    body: response.text(),
+                  })
+                );
+              }
+            })
+            .catch((err) => console.log(err));
+        }
+      }
+    }
+
+    if (removedLinkKeys) {
+      for (let removedLinkKey of removedLinkKeys) {
+        for (let linkData of this.state.linkDataArray) {
+          if (removedLinkKey === linkData["key"]) {
+            fetch(
+              "https://localhost:7009/graph/edges?" +
+                new URLSearchParams({
+                  srcKey: linkData["from"].toString(),
+                  dstKey: linkData["to"].toString(),
+                }),
+              { method: "DELETE" }
+            )
+              .then((response) => {
+                if (response.ok) {
+                  console.log(`Successfully removed edge on backend`);
+                  console.log(linkData);
+                } else {
+                  throw new Error(
+                    JSON.stringify({
+                      status: response.status,
+                      body: response.text(),
+                    })
+                  );
+                }
+              })
+              .catch((err) => console.log(err));
+          }
+        }
+        continue;
```

Comment:
Wait bro I'm so confused why is this continue here? I thought continue skips an iteration of the for loop, but when I tested it out by removing 2 links at the same time the 2nd key wasn't skipped. The code is working I'm just not sure what the continue is for.